server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1631,63 +1636,93 @@ void removeEntitlement(Entitlement entitlement, boolean regenModified) {
         }
 
         // post unbind actions
-        PoolHelper poolHelper = new PoolHelper(this, entitlement);
-        enforcer.postUnbind(consumer, poolHelper, entitlement);
+        for (Entitlement ent : entsToRevoke) {
+            PoolHelper poolHelper = new PoolHelper(this,  ent);
+            enforcer.postUnbind(ent.getConsumer(), poolHelper, ent);
+        }
+
+        List<Entitlement> batch = new ArrayList<Entitlement>();
+        for (int i = 0; i < entsToRevoke.size(); i++) {
+            Entitlement entitlement = entsToRevoke.get(i);
+            batch.add(entitlement);
 
-        if (regenModified) {
-            // Find all of the entitlements that modified the original entitlement,
-            // and regenerate those to remove the content sets.
-            // Lazy regeneration is ok here.
-            this.regenerateCertificatesOf(entitlementCurator.listModifying(entitlement), true);
+            // We work in batches of maximum size 1000.
+            if (i % 1000 == 0) {
+                Set<Entitlement> modifiedEnts = entitlementCurator.batchListModifying(batch);
+                this.regenerateCertificatesOf(modifiedEnts, true);
+                batch.clear();
```

Comment:
Am curious why we are doing this in a batch since all we are doing is marking the entitlement as dirty here ( cause lazy is true), is the reason we are doing this in a batch is that some databases have a limitation of in clause to 1000 ? if yes I need to put similar checks in #1073 . please let me know!